cmd/kops/create_cluster.go

```diff
@@ -942,6 +947,9 @@ func RunCreateCluster(f *util.Factory, out io.Writer, c *CreateClusterOptions) e
 			}
 		}
 	}
+	if len(c.APILoadBalancerSecurityGroups) > 0 && cluster.Spec.API.LoadBalancer == nil {
+		return fmt.Errorf("unable to add security groups to nonexistent API loadbalancer")
```

Comment:
This error handing should be moved into validate.go

Reply:
@chrislovecnm  This error handing is not for validating the cluster spec, it is for cli flags. If you by mistake addd APILoadBalancerSecurityGroups and there is no loadbalancer, it raises the error.

Your thought about a different file for validation is correct. If you run `kops validate cluster`, the error handing should be moved into validate.go. But this for `kops create cluster` with some non-logical or conflicting cli flags.

Reply:
Was not clear enough https://github.com/kubernetes/kops/blob/master/pkg/apis/kops/validation/validation.go

If we choose to have the cli option, we can have it here, but most likely we will either use the option that allows YAML, but we do need validation.go to be updated with the API machinery validation.

Reply:
After removing the cli option, we don't need such error handing. The security groups will be a part of the loadbalancer part of the cluster spec. 

The part that is not clear for me is why do we need validation.go to be updated with the API machinery validation! Could you explain?

Thanks for your time!

Reply:
We do need it in the api validation so we don’t get stuff like empty sg group names ;)

Reply:
The api machinery validation does the same stuff like the cli validation you wrote, but at an api level. The plan is to eventually have kops at a controller, and we need more validation at the api level, and less at the cli level. It eventually ends up at the api anyways. The api is the golang representation of the cluster manifest/spec. If you want to do a separate PR that is fine as well.